src/box/txn.c

```diff
@@ -484,9 +536,14 @@ txn_commit_stmt(struct txn *txn, struct request *request)
 			 * must remain the same
 			 */
 			stmt->does_require_old_tuple = true;
+			if (txn_stmt_decompress_tuples(stmt) != 0)
+				goto fail;
```

Comment:
IMO better put this decompression logic in the on_replace trigger function wrapper instead of the tx manager.

Reply:
Fixed